src/hotspot/share/opto/movenode.cpp

```diff
@@ -185,6 +185,57 @@ CMoveNode *CMoveNode::make(Node *c, Node *bol, Node *left, Node *right, const Ty
   }
 }
 
+// Try to identify min/max patterns in CMoves
+static Node* is_minmax(PhaseGVN* phase, Node* cmov) {
```

Comment:
I'm not a fan of `is_...` methods that do more than a check, but actually have a side-effect.
I also suggest that `cmov` should already have a `CMovNode` type, and there should be an assert here.

I would probably do it similar to `AddNode::IdealIL` and `AddPNode::Ideal_base_and_offset`: call it `CMoveNode::IdealIL_minmax`. But add an assert to check for int or long.

Reply:
And then you could actualy move the call to `CMoveNode::Ideal`.

Reply:
Who knows, maybe we one day extend this to other types

Reply:
I think moving the call to `CMoveNode::Ideal` would be a good idea, since it de-duplicates the call site. Would it still assert on non-supported types, then? I think it may make more sense if it simply filtered out the cmov types that it doesn't (currently) support.